pkg/controller/datasource-controller.go

```diff
@@ -237,20 +237,9 @@ func NewDataSourceController(mgr manager.Manager, log logr.Logger, installerLabe
 }
 
 func addDataSourceControllerWatches(mgr manager.Manager, c controller.Controller, log logr.Logger) error {
```

Comment:
This function is getting a bit messy and overly large, IMO would benefit from a small refactor (at least grouping watches and indexers together). Not a must for this PR, your changes look great.